### PR TITLE
Share retry after across independent execution contexts

### PR DESCRIPTION
**Pull request by KnightNiwrem**
If 429 is triggered by some API call in a middleware or some other independent setTimeout call, other async threads that uses the same sessionKey (i.e. bot api token or id) shall also be aware of the recalculated retry_after value.

This attempts to ensure the 429 -> stop all calls until.. -> start sending model.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -43,6 +46,16 @@ export interface AutoRetryOptions {
      * (https://en.m.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_server_errors)
      */
     retryOnInternalServerErrors: boolean
+    /**
+     * All network requests that are sent out from the same session should pause
+     * if one of the request receives a 429 response from Telegram.
+     *
+     * Note: Session, for transformers, are from the Bot's perspective to
+     * Telegram. This is unlike middlewware sessions, which are from user
```
Comment (KnightNiwrem) [inline src/index.ts]: Typo tsk

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
Rate limits are enforced on a per-method basis. They may affect each other indirectly, but we should delay all calls to the same method—not session key.

Amazing work btw!

**Comment by KnightNiwrem** [comment]
> Rate limits are enforced on a per-method basis. They may affect each other indirectly, but we should delay all calls to the same method—not session key.
> 
> Amazing work btw!

Fixed.. I think?

**Comment by KnorpelSenf** [comment]
@KnightNiwrem how well did you test this?

**Comment by KnorpelSenf** [comment]
This is not a good idea. For examples, responses to messages might still be sent while broadcasting is slowed down. It makes sense to keep unrelated API call independent, and not apply rate limits of one call to other calls.

Bot developers need to make sure that they don't call a method too many times concurrently.

**Pull request closed without merging**